lib/endpoints/class-wp-rest-posts-terms-controller.php

```diff
@@ -0,0 +1,229 @@
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+		) );
+
+		register_rest_route( 'wp/v2', '/' . $base . '/(?P<id>[\d]+)/terms/(?P<taxonomy>[^/]+)/(?P<term_id>[\d]+)', array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			),
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'create_item' ),
+				'permission_callback' => array( $this, 'create_item_permissions_check' ),
+			),
+			array(
+				'methods'         => WP_REST_Server::DELETABLE,
+				'callback'        => array( $this, 'delete_item' ),
+				'permission_callback' => array( $this, 'create_item_permissions_check' ),
+			),
+		) );
+	}
+
+	/**
+	 * Get a collection of posts
+	 *
+	 * @param WP_REST_Request $request Full details about the request
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+
+		$post = get_post( $request['id'] );
+		$taxonomy = $request['taxonomy'];
+
+		if ( empty( $post->ID ) || $this->post_type !== $post->post_type ) {
+			return new WP_Error( 'rest_post_invalid_id', __( 'Invalid post ID.' ), array( 'status' => 404 ) );
+		}
+
+		$valid_taxonomies = get_object_taxonomies( $post->post_type );
+		if ( ! in_array( $taxonomy, $valid_taxonomies ) ) {
+			return new WP_Error( 'rest_post_taxonomy_invalid', __( 'Invalid taxonomy for post ID.' ), array( 'status' => 404 ) );
+		}
+
+		$terms = wp_get_object_terms( $post->ID, $taxonomy );
+		$terms_controller = new WP_REST_Terms_Controller( $request['taxonomy'] );
+
+		$response = array();
+		foreach ( $terms as $term ) {
+			$data = $terms_controller->prepare_item_for_response( $term, $request );
+			$response[] = $this->prepare_response_for_collection( $data );
+		}
+
+		$response = rest_ensure_response( $response );
+
+		return $response;
+	}
+
+	/**
+	 * Get a single post
+	 *
+	 * @param WP_REST_Request $request Full details about the request
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_item( $request ) {
+		$post     = get_post( $request['id'] );
+		$taxonomy = $request['taxonomy'];
+		$term_id  = absint( $request['term_id'] );
+
+		if ( empty( $post->ID ) || $this->post_type !== $post->post_type ) {
+			return new WP_Error( 'rest_post_invalid_id', __( 'Invalid post ID.' ), array( 'status' => 404 ) );
+		}
+
+		$valid_taxonomies = get_object_taxonomies( $post->post_type );
+		if ( ! in_array( $taxonomy, $valid_taxonomies ) ) {
+			return new WP_Error( 'rest_post_taxonomy_invalid', __( 'Invalid taxonomy for post ID.' ), array( 'status' => 404 ) );
+		}
+
+		if ( ! get_term_by( 'term_taxonomy_id', $term_id, $taxonomy ) ) {
```

Comment:
As this requires setting up a new Request object, I don't know how pretty that's going to be